### PR TITLE
Crucifixes and Warding Off Curses

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

New item; crucifix, a limited supply neck item that protects against evil spirits and curses. Available to the chaplain witch hunter set.
Also, rosary beads; works similar to crucifix, but can be destroyed with fire. available at the chaplain vendor and religious supply packs.

## Why It's Good For The Game

There is really no counter to curses, as they ignore magic immunity. Just adds a bit to complexity to curses.

## Changelog
:cl:
add: Added crucifixes to chaplain's witch hunter set, which wards against curses.
add: Added rosary beads, which work similarly to crucifixes; available at chaplain vendor and religious supply packs in cargo.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm [hunk 1]
```diff
@@ -158,6 +158,10 @@
 	var/chosen_mob = input("Select the person you wish to curse","Your target") as null|anything in sortList(compiled_list, /proc/cmp_mob_realname_dsc)
 	if(!chosen_mob)
 		return FALSE
+	var/mob/living/living_mob = chosen_mob
+	if (istype(living_mob) && HAS_TRAIT(chosen_mob, TRAIT_WARDED))
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]:
```suggestion
	if (istype(living_mob) && HAS_TRAIT(living_mob, TRAIT_WARDED))
```

Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: oh shit

**Comment by Fronsis** [comment]
Does it also helps on removing curses like the animal mask?

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm [hunk 2]
```diff
@@ -179,3 +179,29 @@
...
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "crucifix"
+	icon_state = "eye_medalion"
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/clothing/neck/crucifix/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(slot == SLOT_NECK && istype(user))
+		ADD_TRAIT(user, TRAIT_WARDED, CLOTHING_TRAIT)
+		user.update_sight()
```
Comment (PowerfulBacon) [inline code/modules/antagonists/eldritch_cult/eldritch_items.dm]: why is sight being updated?

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm [hunk 3]
```diff
@@ -179,3 +179,29 @@
...
+
+/obj/item/clothing/neck/crucifix/dropped(mob/user)
```
Comment (PowerfulBacon) [inline code/modules/antagonists/eldritch_cult/eldritch_items.dm:197]: Is this called on dequip?

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm [hunk 4]
```diff
@@ -179,3 +179,29 @@
...
 	icon_state = "eldrich_flask"
 	list_reagents = list(/datum/reagent/eldritch = 50)
+
+/obj/item/clothing/neck/crucifix
+	name = "crucifix"
+	desc = "In the eventuality that one of those you falesly accused is, in fact, a real witch, this will ward you against their curses."
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "crucifix"
+	icon_state = "eye_medalion"
```
Comment (PowerfulBacon) [inline code/modules/antagonists/eldritch_cult/eldritch_items.dm]: double icon state

**Review by PowerfulBacon — COMMENTED** [review]
Should probably account for the barnyard curse, although doesn't really matter. Also needs a icon state while being worn.

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm (hunk 2 above)
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_items.dm]: > Also needs a icon state while being worn.

Gonna have to nag spriters for this one.

**Comment by KazooBard** [comment]
Idea for rosary: usable in hand for bigger channeled prayer text.

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm (hunk 3 above)
Comment (park66665) [reply code/modules/antagonists/eldritch_cult/eldritch_items.dm:197]: `dropped` is called at `doUnEquip` and `put_in_hands`. It even is called on `Destroy` via `temporarilyRemoveItemFromInventory`.

File: code/modules/vending/wardrobes.dm [hunk 5]
```diff
@@ -382,8 +382,9 @@
 					/obj/item/clothing/head/medievaljewhat = 3,
 					/obj/item/clothing/suit/chaplainsuit/clownpriest = 1,
 					/obj/item/clothing/head/clownmitre = 1,
-					/obj/item/clothing/neck/cloak/chap/bishop = 1)
-	premium = list(/obj/item/clothing/suit/chaplainsuit/bishoprobe = 1,
+          /obj/item/clothing/neck/cloak/chap/bishop = 1)
+	premium = list(/obj/item/clothing/suit/chaplainsuit/bishoprobe = 1,	
+					/obj/item/clothing/neck/crucifix/rosary = 1,
```
Comment (park66665) [inline code/modules/vending/wardrobes.dm]:
```suggestion
					/obj/item/clothing/neck/cloak/chap/bishop = 1)
	premium = list(/obj/item/clothing/suit/chaplainsuit/bishoprobe = 1,
					/obj/item/clothing/neck/crucifix/rosary = 1,
```

**Review by park66665 — APPROVED** [review]
All clear. This PR is good to go.

**Pull request merged**